### ISSUE TITLE
BUG: save broken

### ISSUE THREAD

**Issue by pburdet** [hyperspy/hyperspy#695]
```save``` does not work with old .hdf5 files. Due to the type of string, I guess.

```python 

s = hs.datasets.example_signals.EDS_SEM_Spectrum()
s.save('test')

```

```python 

s = hs.signals.Signal(range(10))
s.metadata.set_item('result',[u'Al', u'Cu'])
s.save('a.hdf5', overwrite=True)

```

**Comment by pburdet** [comment]
Also ```save``` becomes way slower saving list of signals.

With 0.8

```python 

>>> %%time
>>> s = hs.utils.stack([hs.utils.example_signals.EDS_SEM_Spectrum()]*100)
>>> s.add_elements(['Al', 'C', 'Cu', 'Mn', 'Zr'])
>>> s.metadata.set_item('result', s.get_lines_intensity())
>>> s.save('test', overwrite=True)
The test.hdf5 file was created
Wall time: 3.05 s

```


With 0.8.1

```python 

>>> %%time
>>> s = hs.stack([hs.datasets.example_signals.EDS_SEM_Spectrum()]*100)
>>> s.add_elements(['Al', 'C', 'Cu', 'Mn', 'Zr'])
>>> s.metadata.set_item('result', s.get_lines_intensity())
>>> s.save('test', overwrite=True)
The test.hdf5 file was created
Wall time: 40.9 s

```

**Comment by to266** [comment]
Regarding the first bug, would you rather keep the strings as unicode and (if you have a very long list of them) save them very very slowly, or convert it to `str` and save it much quicker, but have to deal with the unicode->str conversion at some point if you have any fancy characters?

**Comment by to266** [comment]
I'll fix the slow thing as well, should be just as quick as previous

**Comment by pburdet** [comment]
I'm fine with the second solution. But a warning would be very welcome. I don't understand why at some point hyperspy was saving as unicode characters that could have been saved as as ```str```.

**Comment by to266** [comment]
Previously we saved a list "artificially", but that's very slow with long lists. Now we try to use numpy (i.e. we change lists to numpy arrays) to save much much quicker, but then saving the numpy arrays constrains the dtypes that hdf5 can save. The slow saving was due to
```python
np.array([s,s,s])
```
not returning any errors and instead creating something horrible... And that took a long time with many signals in a list. So now (i.e. in the fix I'll push in 1hour or so, after I've had some food in me) we first check if there are any signals in a list, and only try to "numpy" things if there aren't any.

Anyway, thanks for the bugreports. I wish you noticed them yesterday, as I was just compiling the bundles as you opened this issue, but no worries now :)
